src/analysis/mod.rs

```diff
@@ -69,7 +71,47 @@ pub fn run(env: &mut Env) {
     }
     if !to_analyze.is_empty() {
         error!("Not analyzed {} objects due unfinished dependencies", to_analyze.len());
+        return;
     }
+
+    // Analyze free functions as the last step once all types are analyzed
+    analyze_global_functions(env);
+}
+
+fn analyze_global_functions(env: &mut Env) {
+    let ns = env.library.namespace(library::MAIN_NAMESPACE);
+
+    let full_name = format!("{}.*", ns.name);
+
+    let obj = match env.config.objects.get(&*full_name) {
+        Some(obj) => obj,
+        None => return,
+    };
+
```

Comment:
IMHO better `check obj.status.need_generate()` before processing.
This way we can disable generation without deleting object from Gir.toml